documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,277 @@
+ Removing this field means having no metrics exposed.
+ 
+An example of cluster configuration ConfigMap is the following.
+
+.Kafka cluster ConfigMap
+[source,yaml,options="nowrap"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cluster
+  labels:
+    strimzi.io/kind: cluster
+    strimzi.io/type: kafka
+data:
+  kafka-nodes: "3"
+  kafka-image: "strimzi/kafka:latest"
+  kafka-healthcheck-delay: "15"
+  kafka-healthcheck-timeout: "5"
+  zookeeper-nodes: "1"
+  zookeeper-image: "strimzi/zookeeper:latest"
+  zookeeper-healthcheck-delay: "15"
+  zookeeper-healthcheck-timeout: "5"
+  KAFKA_DEFAULT_REPLICATION_FACTOR: "3"
+  KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: "3"
+  KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: "3"
+  kafka-storage: |-
+    { "type": "ephemeral" }
+  zookeeper-storage: |-
+    { "type": "ephemeral" }
+  kafka-metrics-config: |-
+    {
+      "lowercaseOutputName": true,
+      "rules": [
+          {
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*><>Count",
+            "name": "kafka_server_$1_$2_total"
+          },
+          {
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*, topic=(.+)><>Count",
+            "name": "kafka_server_$1_$2_total",
+            "labels":
+            {
+              "topic": "$3"
+            }
+          }
+      ]
+    }
+  zookeeper-metrics-config: |-
+    {
+      "lowercaseOutputName": true
+    }
+----
+
+==== Storage
+
+Kafka needs to save information like data logs (so received messages), consumer offsets and so on in some way; the same is true for 
+Zookeeper about znodes children and values.
+The Strimzi project allows to save such a data in an "ephemeral" way (using `emptyDir`) or in a "persistent-claim" way using persistent 
+volumes.
+It's possible to provide the storage configuration in the related ConfigMap using a JSON string as value for the 
+`kafka-storage` and `zookeeper-storage` fields. It's important to mention that such a storage configuration can't be changed
+when the cluster is up and running.
+
+The JSON representation has a mandatory `type` field for specifying the type of storage to use ("ephemeral" or "persistent-claim").
+In case of "persistent-claim" type the following fields can be provided as well :
+
+* `size`: defines the size of the persistent volume claim (i.e 1Gi)
+* `class` : the Kubernetes/OpenShift storage class to use for dynamic volume allocation
+* `selector`: allows to select a specific persistent volume to use. It contains a `matchLabels` field which defines an 
+inner JSON object with key:value representing labels for selecting such a volume.
+* `delete-claim`: specifies if the persistent volume claim has to be deleted when the cluster is un-deployed. Default is `false`
+
+The "ephemeral" storage is really simple to configure and the related JSON string is something like this.
```

Comment:
Above you mentioned ephemeral storage first and then persistent. But here you've gone into some details about persistent, and now you're back to ephemeral. I think it would be clearer if you give the complete low-down on ephemeral first, with example, and then go into the details about persistent, with examples.